src/fhirtypes/common.ts

```diff
@@ -176,3 +178,24 @@ export function replaceField(
     }
   }
 }
+
+/**
+ * Checks if the provided string matches the regular expression specified
+ * in the invariant for "name" properties. A name must be between 1 and 255 characters long,
+ * begin with an uppercase letter, and contain only uppercase letter, lowercase letter,
+ * numeral, and '_' characters.
+ *
+ * @see {@link http://hl7.org/fhir/R4/structuredefinition-definitions.html#StructureDefinition.name}
+ * @see {@link http://hl7.org/fhir/R4/valueset-definitions.html#ValueSet.name}
+ * @see {@link http://hl7.org/fhir/R4/codesystem-definitions.html#CodeSystem.name}
+ * @param {string} name - The name to check against the name invariant
+ * @throws {InvalidNameError} when the name does not match the regular expression
+ */
+
+export function validateFHIRName(name: string): void {
+  if (!nameRegex.test(name)) {
+    throw new InvalidNameError(name);
+  }
+}
+
+const nameRegex = /^[A-Z]([A-Za-z0-9_]){0,254}$/;
```

Comment:
Organizationally, I think it might make sense to put this above the function that uses it.  I wasn't going to say anything, except that I noticed for the `validateFHIRId`, you _did_ put it above the function.  So might as well be consistent and I think above is the way to go!